doc/internals.rst

```diff
@@ -231,3 +231,200 @@ re-open it directly with Zarr:
     zgroup = zarr.open("rasm.zarr")
     print(zgroup.tree())
     dict(zgroup["Tair"].attrs)
+
+
+How to add a new backend
+------------------------------------
+
+Adding a new backend for read support to Xarray is easy, and does not require
+to integrate any code in Xarray; all you need to do is approaching the
+following steps:
+
+- Create a class that inherits from Xarray py:class:`~xarray.backend.commonBackendEntrypoint`
+- Implement the method ``open_dataset`` that returns an instance of :py:class:`~xarray.Dataset`
+- Declare such a class as an external plugin in your setup.py.
```

Comment:
Can we link to the entrypoints documentation here.

```suggestion
- Declare such a class as an external plugin in your ``setup.py``.
```

EDIT: I see it's linked below. Won't hurt to add a link here also.